src/preset_cli/cli/superset/sync/dbt/lib.py

```diff
@@ -198,6 +201,31 @@ def build_snowflake_sqlalchemy_params(target: Dict[str, Any]) -> Dict[str, Any]:
     return parameters
 
 
+def create_sqlalchemy_engine(url: URL):
+    """
+    Returns a SQLAlchemy engine or raises an error if missing required dependency.
+    """
+    try:
+        return create_engine(url)
+    except NoSuchModuleError as exc:
+        string_url = str(url)
+        dialect = string_url.split("://", maxsplit=1)[0]
+        # TODO: Handle more DB engines that require an additional package
+        if dialect == "snowflake":
+            raise CLIError(
+                (
+                    "Missing required package. "
+                    'Please run ``pip install "preset-cli[snowflake]"`` to install it.'
+                ),
+                1,
+            ) from exc
```

Comment:
In the future this could be something like:
``` python
if dialect in {"snowflake", "redshift", "bigquery"}:
    raise CLIError(
        (
            "Missing required package. "
            f'Please run ``pip install "preset-cli[{dialect}]"`` to install it.'
        ),
        1,
    ) from exc
```